week-3/quote-generator/quotes.js

```diff
@@ -490,4 +494,58 @@ const quotes = [
   },
 ];
 
+let autoPlayInterval; // Variable to store the interval ID
+
+// Function to start auto-generate
+function startAutoGenerate() {
+  autoPlayInterval = setInterval(() => {
+    displayRandomQuote();
+  }, 60000); // Change 60000 to 5000 for testing every 5 seconds
+}
+
+// Function to stop auto-generate
+function stopAutoGenerate() {
+  clearInterval(autoPlayInterval);
+}
+
+// Function to display a random quote
+function displayRandomQuote() {
+  const randomQuote = getRandomQuote();
+  displayQuote(randomQuote);
+}
+
+// Function to display a specific quote
+function displayQuote(quote) {
+  const quoteElement = document.getElementById("quote");
+  const authorElement = document.getElementById("author");
+
+  quoteElement.textContent = quote.quote;
+  authorElement.textContent = quote.author;
+}
+
+// Event listener for the "New quote" button
+document.getElementById("new-quote").addEventListener("click", () => {
+  displayRandomQuote();
+});
+
+// Event listener for the auto-play toggle
+const autoPlayToggle = document.getElementById("auto-play-toggle");
+
+autoPlayToggle.addEventListener("change", () => {
+  if (autoPlayToggle.checked) {
+    startAutoGenerate();
+    document.getElementById("auto-play-status").textContent = "auto-play: ON";
+  } else {
+    stopAutoGenerate();
+    document.getElementById("auto-play-status").textContent = "auto-play: OFF";
+  }
+});
+
+// Initial display of a random quote
+displayRandomQuote();
+
+function getRandomQuote() {
+  return pickFromArray(quotes);
+}
+
```

Comment:
And here you're re-declaring this function again which means you replace the previous one from line `23`.